DiegoPino/islandora_internet_archive_bookreader
# Bookreader 2.x with islandora_internet_archive_bookreader module 7.x01.11+[HEAD]

**JIRA Ticket**: (https://jira.duraspace.org/browse/ISLANDORA-1947)

# What does this Pull Request do?

* Switch islandora_internet_archive_bookreader to use the 2.0.2 version of the bookreader library. This provides an updated theme, a number of bug fixes, and usability improvements. It also eliminates the need to maintain our own customized fork of the bookreader library.

# What's new?

* Brings Diego's 7.x branch up to date with islandora upstream.
* Replaces dependency on [our fork of the Internet Archive BookReader](https://github.com/Islandora/internet_archive_bookreader.git) with [the 2.0.2 version from internetarchive.org](https://github.com/internetarchive/bookreader/tree/2.0.2). *(It has not been tested with more recent 2.x versions, or 3.x)*
* Removes bundled copies of jquery (1.5) and jquery_ui (1.8.10). They are not needed.
* Extensively modifies islandora_book_reader.js to work with the 2.x version of bookreader, and provide Islandora-specific customization that had been part of our fork of the bookreader library (credit: Diego and Giancarlo!).
* Removes some additional features that are in Diego's 7.x-2dev branch which are not strictly necessary to support the newer bookreader library (I propose that these features be evaluated separately):
  - Default image width
  - Info tab view
  - Mobile header
* Discards some changes that had been introduced to support IIIF usage but are not needed because we are working from the current 7.x branch.

# How should this be tested?

Because this is a large change to an existing feature, testing should be pretty broad.

* Install/upgrade per the README file. (Clear all caches if upgrading.)
* Configure
  - Consult the README section on configuring this module. Do all the controls work as described, save their settings, etc?
* Navigate to a multiple page book object (create one if necessary) and test display and functionality:
  - Top of the viewer controls:
    - Text popup (shows the text of the currently displayed pages in a popup, if available).
    - Info popup (shows metadata in a popup).
    - Sharing popup (shows social media and mail sharing links).
    - Zoom in and out.
    - Search inside the book. Pick a word that you know occurs several times within the book. 
    - Full screen display.
  - Bottom of the viewer controls:
    - One-up, two-up and tiled display. Do the pages display correctly, showing edge to edge on all four sides?
    - Forward and back buttons.
    - Show/hide controls button.
- Testing is needed for both djatoka and IIIF implementations. The main trick for IIIF is making sure you have the IIIF Identifier set correctly. 

# Additional Notes:

* Does this change require documentation to be updated?
  *Yes. This PR includes a minor change to the README regarding how to install the bookreader library.*
* Does this change add any new dependencies?
  *No.*
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)?
  *No.*
* Could this change impact execution of existing code?
  *No. This only affects the display of book objects.*

# Interested parties
@DiegoPino @giancarlobi
